### PR TITLE
Prepare 1.6.1.1 (only hackage release)

### PR DESCRIPTION
**Pull request by jneira**
* We will use the fourth position for this release cause it will not have a github/gitlab/ghcup release, only a hackage one
* It tries to fix the build workflow, broken since #2668 (see https://github.com/haskell/haskell-language-server/actions/runs/1779913965)
  * I think the cause has been mix the dynamic build in the test workflow and the static one in the build one
  * The cabal store restored from the cache, populated by test/caching workflow makes the static build fail
  * The solution is moving the flag setting to the generic action used for all workflows
* This will be backported to the 1.6.1.1-hackage branch, created on the 1.6.1.0 one

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2681"></a>

### CODE REVIEW

**Comment by jneira** [comment]
@Mergifyio backport 1.6.1.1-hackage

**Review by pepeiborra — APPROVED** [review]

File: .github/actions/setup-build/action.yml [hunk 1]
```diff
@@ -63,10 +63,15 @@ runs:
         fi
       shell: bash
 
+    - name: Disable -dynamic
```
Comment (michaelpj) [inline .github/actions/setup-build/action.yml:66]: I still think this needs a comment.

Comment (jneira) [reply .github/actions/setup-build/action.yml:66]: there is one in the .cabal file, but maybe could be refer it here

Comment (jneira) [reply .github/actions/setup-build/action.yml:70]: finally added a full comment with ci specific caveats

File: .github/actions/setup-build/action.yml [hunk 2]
```diff
@@ -63,6 +63,10 @@ runs:
         fi
       shell: bash
 
+    # The default build for haskell-language-server executable is dynamic for linux and macOS
```
Comment (michaelpj) [inline .github/actions/setup-build/action.yml:66]: Great, thank you.

**Review by michaelpj — APPROVED** [review]

**Comment by jneira** [comment]
I changed the title of the troubleshooting section, breaking the url of the th warning. I had to correct it in the original pr
To have hardcoded urls in the code is a real PITA

**Comment by michaelpj** [comment]
> To have hardcoded urls in the code is a real PITA

Maybe it should just link to the documentation root and then say in prose where to look? e.g. "Check the [documentation](<link>), look for BLAH in the troubleshooting section"

**Comment by jneira** [comment]
> > To have hardcoded urls in the code is a real PITA
> 
> Maybe it should just link to the documentation root and then say in prose where to look? e.g. "Check the [documentation](<link>), look for BLAH in the troubleshooting section"

buy it would be longer and less direct... a  more specific and dedicated URL + redirection in docs would serve as the needed indirection, rtd allows those things?

**Comment by jneira** [comment]
https://hackage.haskell.org/package/haskell-language-server-1.6.1.1 released

**Pull request merged**